### ISSUE TITLE
Icons within buttons should have aria-hidden="true" on the SVG

### ISSUE THREAD

**Issue by emyarod** [IBM/carbon-components-react#1745]
Related: https://github.com/IBM/carbon-components/issues/1630

> The icon in a ["Button With icon"](https://www.carbondesignsystem.com/components/button/code) is purely decorative, and as such, it (the svg) should have `aria-hidden="true"` so that screen readers don't try to say something about them.
> 
> Please add `aria-hidden="true"` to the svg element for all of the "Xxx Button With icon" examples on the Carbon [Button component](https://www.carbondesignsystem.com/components/button/code) page.
> 
> It might be nice if there was an "Accessibility Notes" section on the Button Component page that mentioned this, i.e. something like:
> - Note: The SVG in a "Button With icon" is purely decorative, and so it should have aria-hidden="true" to hide it from Assistive Technology such as screen readers. It does not need to have a title or desc.
